SlackNet.Bot/SlackBot.cs

```diff
@@ -239,8 +240,17 @@ public void AddOutgoingMiddleware(Func<IObservable<BotMessage>, IObservable<BotM
         /// </summary>
         public IObservable<IMessage> Messages => _incomingMessages.AsObservable();
 
-        private async Task<SlackMessage> CreateSlackMessage(MessageEvent message) =>
-            new SlackMessage(this)
+        private async Task<SlackMessage> CreateSlackMessage(MessageEvent message)
+        {
+            SlackMessage result;
+
+            if (message.User == null && (message is SlackNet.Events.BotMessage b))
+            {
+                BotInfo botInfo = await GetBotUserIdById(b.BotId);
```

Comment:
will do if you really want.

I have a best practice for this: use var only iff the line already contains the explicit type, so the code is readable and understandable without the need of an IDE.

In my team, we banned var completely, because there are cases where a right hand side typo will be hidden compelling and transforms to a runtime error later (e.g. create base class instead of subclass, use another implementation than expected of the same interface etc...